sympy/solvers/ode/ode.py

```diff
@@ -1079,10 +1079,10 @@ class in it.  Note that a hint may do this anyway if
             point = boundary.get('f0', 0)
             value = boundary.get('f0val', C1)
             check = cancel(r[d]/r[e])
-            check1 = check.subs({x: point, y: value})
+            check1 = check.subs({x: point, y: value}).simplify()
             if not check1.has(oo) and not check1.has(zoo) and \
                 not check1.has(nan) and not check1.has(-oo):
-                check2 = (check1.diff(x)).subs({x: point, y: value})
+                check2 = check.diff(x).subs({x: point, y: value}).simplify()
```

Comment:
Why is `simplify` added here?

Reply:
Considering the test case:

```python
bcs = {f(x).subs(x, oo): C1, f(x).diff(x).subs(x, oo): C2}
eq2 = f(x).diff(x) + 4 * C3 * f(x) / (x * (C3 + C4))
```

At line 1082, if `simplify` is not used, the expression stays as `4*C1*K1/(oo*K1 + oo*K2)` and hence does not pass the next if check. Both `simplify` calls are added to evaluate the substituted expressions at `x=oo`.

At line 1085, without `simplify`, the expression stays as `4*C1*K1*(-K1 - K2)/(oo*K1 + oo*K2)**2`

Reply:
@oscarbenjamin

Is there something else that would be better to use here than `simplify`?